cmd/singularity/build_test.go

```diff
@@ -996,7 +1039,19 @@ func verifyEnv(t *testing.T, imagePath string, env []string, flags []string) err
 	}
 	args = append(args, imagePath, "env")
 
+	// We always prefer to run tests with a clean temporary image cache rather
+	// than using the cache of the user running the test.
+	// In order to unit test using the singularity cli that is thread-safe,
+	// we prepare a temporary cache that the process running the command will
+	// use.
+	tmpImgCache, err := ioutil.TempDir("", "image-cache-")
```

Comment:
this bit of code is also repeated a couple of times...

Maybe:

```
func setupCmdCache(t *testing.T, cmd *exec.Cmd, tag string) {
    dir, err := ioutil.TempDir("", tag + "-")
    if err != nil {
        t.Fatalf("...")
    }
    env := cache.DirEnv + "=" + dir
    if cmd.Env == nil {
        cmd.Env = os.Environ
    }
    cmd.Env = append(cmd.Env, env)
}
```

and then you do:

```
    cmd := exec.Command(cmdPath, args...)
    setupCmdCache(t, cmd, "image-cache")
    out, err := cmd.CombinedOuptut()
```